src/doc/doc-builders.js

```diff
@@ -1,5 +1,9 @@
 "use strict";
 
+/**
+ * @typedef {string | Object} Doc
```

Comment:
My testing shows that TS treats both `object` and `Object` as `any` is JSDoc. So `string | Object` is `any` too. Any idea how this can be improved without introducing an extra `.d.ts` file?

Reply:
Yea... instead if `object` you should define it
like `@typedef {string | {type: string, ...}} Doc`

Reply:
> like `@typedef {string | {type: string, ...}} Doc`

Did not quite work in JSDoc, I fixed it a different way.

I confirmed that if I try the following:

```diff
diff --git a/src/doc/doc-builders.js b/src/doc/doc-builders.js
index a35d1597..aa56ffd0 100644
--- a/src/doc/doc-builders.js
+++ b/src/doc/doc-builders.js
@@ -2,7 +2,7 @@

 /**
  * @typedef {Object} DocObject
- * @property {string} type
+ * @property {string} tpe
  *
  * @typedef {string | DocObject} Doc
  */
```

then I would get a bunch of type errors from `tsc`.